File: python/primegen.py
"""
various functions to generate primes
"""
def prime_by_trial():
    #by trial
    primes = [2]
    if len(primes) == 1:
            yield primes[-1]
    while True:
        x = primes[-1] + 1
        is_prime = False
        while not is_prime:
            is_prime = True
            for prime in primes:
                if (x % prime == 0):
                    is_prime = False
                    break
            if (is_prime):
                #print("Prime: {}\t length: {}\n".format(x, len(primes) + 1))
                primes.append(x)
            else:
                x += 1
        yield primes[-1]

def primes_by_sieve(max_num):
    #uses a sieve to find all primes below big_num, returns all primes
    #why it runs so slowly I'm not sure...
    nums = [True] * int(max_num)
    primes = []
    for i in range(2, len(nums)):
        if nums[i]:
            primes.append(i)
            j = 2
            while (j * i) < len(nums):
                nums[j * i] = False
                j += 1
    return primes

File: julia/primegen.jl
function primes_by_sieve(max_num::Int)::Vector{Int}

    nums = trues(max_num)
    primes = []
    
    for i in 2:max_num
        if nums[i]
            push!(primes, i)
            j = 2
            while (i * j) < length(nums)
                nums[i * j] = false
                j += 1
            end
        end
    end
    primes
end

function primes_by_trial()
    primes = [2]
    Channel(ctype=Int, csize=1) do c
        while (true)
            put!(c, last(primes))
            candidate = last(primes) + 1
            while 0 in (candidate .% primes)
                candidate += 1
            end
            push!(primes, candidate)
        end
    end
 end